src/metaopt/io/parsing.py

```diff
@@ -0,0 +1,257 @@
+        '-v', '--verbose',
+        action='count', default=0,
+        help="logging levels of information about the process (-v: INFO. -vv: DEBUG)")
+
+    moptgroup = parser.add_argument_group(
+        "MetaOpt arguments (optional)",
+        description="These arguments determine mopt's behaviour")
+
+    moptgroup.add_argument(
+        '-e', '--exp-name',
+        type=str, metavar='stringID',
+        help="experiment's unique name; "
+             "use an existing name to resume an experiment "
+             "(default: %s)" % DEF_CMD_EXP_NAME[1])
+
+    moptgroup.add_argument(
+        '--max-trials', type=int, metavar='#',
+        help="number of jobs/trials to be completed "
+             "(default: %s)" % DEF_CMD_MAX_TRIALS[1])
+
+    moptgroup.add_argument(
+        "--pool-size", type=int, metavar='#',
+        help="number of concurrent workers to evaluate candidate samples "
+             "(default: %s)" % DEF_CMD_POOL_SIZE[1])
+
+    moptgroup.add_argument(
+        '-mc', '--moptconfig',
+        type=argparse.FileType('r'), metavar='path-to-config',
+        help="user provided mopt configuration file")
+
+    usergroup = parser.add_argument_group(
+        "User script related arguments",
+        description="These arguments determine user's script behaviour "
+                    "and they can serve as metaopt's parameter declaration.")
+
+    usergroup.add_argument(
+        '-c', '--userconfig',
+        type=str, metavar='path-to-config',
+        help="your script's configuration file (yaml, json, ini, ...anything)")
+
+    usergroup.add_argument(
+        'userscript', type=str, metavar='path-to-script',
+        help="your experiment's script")
+
+    usergroup.add_argument(
+        'userargs', nargs=argparse.REMAINDER, metavar='...',
+        help="command line arguments to your script (if any)")
+
+    args = vars(parser.parse_args())  # convert to dict
+    # Explicitly add metaopt's version as experiment's metadata
+    args['version'] = metaopt.__version__
+    moptfile = args.pop('moptconfig')
+    config = dict()
+    if moptfile:
+        config = yaml.safe_load(moptfile)
+
+    if args['verbose'] == 1:
+        logging.basicConfig(level=logging.INFO)
+    elif args['verbose'] == 2:
+        logging.basicConfig(level=logging.DEBUG)
+
+    return args, config
+
+
+def default_options(user, starttime):
+    """
+    Create a nesteddict with options from the default configuration
+    files, respecting precedence from application's default, to system's and
+    user's.
+
+    .. seealso:: `DEF_CONFIG_FILES_PATHS`
+
+    """
+    defcfg = nesteddict()
+
+    # get some defaults
+    defcfg['exp_name'] = DEF_CMD_EXP_NAME[1].format(
+        user=user,
+        starttime=starttime.isoformat())
+    defcfg['max_trials'] = DEF_CMD_MAX_TRIALS[0]
+    defcfg['pool_size'] = DEF_CMD_POOL_SIZE[0]
+
+    # get default options for some managerial variables (see `ENV_VARS`)
+    for signif, evars in six.iteritems(ENV_VARS):
```

Comment:
Significant seems confusing. At least if confused but me and Fred. Any other idea how to name this?

Reply:
Also `signif` and `evars` names needs to much though to decrypt. `significant` and `env_vars` are more explicit.

Reply:
I thought of this as `signifier`. I took note of that, I will try to make more descriptive names in general.